GOWINDA="/Volumes/Temp/Lukas/Tools/gowinda/Gowinda-1.12.jar"
#GTF="/Volumes/Temp/Lukas/reference_genome/mel/dmel5.38-clean_gowinda.gtf"
GTF="/Volumes/Temp/Lukas/reference_genome/mel/dmel-all-no-analysis-r5.52_flybase_gowinda.gtf"
GTF_GO="/Volumes/Temp/Lukas/reference_genome/mel/dmel-all-no-analysis-r5.52_flybase_gowinda_go_func_assoc_sync.gtf"
GTF_CESAR="/Volumes/Temp/Lukas/reference_genome/mel/dmel-all-no-analysis-r5.52_flybase_gowinda_cesar_sync.gtf"
GTF_KEGG="/Volumes/Temp/Lukas/reference_genome/mel/dmel-all-no-analysis-r5.52_flybase_gowinda_kegg_sync.gtf"
GTF_REACTOME="/Volumes/Temp/Lukas/reference_genome/mel/dmel-all-no-analysis-r5.52_flybase_gowinda_reactome_sync.gtf"
GO="/Volumes/Temp/Lukas/reference_genome/mel/dro_mel_funcassociate_go_associations.txt"
GO_250="/Volumes/Temp/Lukas/Kegg/dro_mel_funcassociate_go_associations_lt_250genes_per_cat.txt"
GO_300="/Volumes/Temp/Lukas/Kegg/dro_mel_funcassociate_go_associations_lt_250genes_per_cat.txt"
KEGG="/Volumes/Temp/Lukas/Kegg/kegg_pathways_flybase.gowinda"
REACTOME="/Volumes/Temp/Lukas/Kegg/reactome_pathways_flybase.gowinda"
RAY_ASSOX="/Volumes/Temp/Lukas/Kegg/associations_CESARgenes_flybase.txt"
GO=$GO_250
TOTAL=$1
SIGN=$2
 nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $GO --annotation-file $GTF_GO --simulations 100000 --min-significance 1 --gene-definition gene --threads 2 --output-file `basename $SIGN .cmhout`"_gene_GO.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_gene_GO.gowinda".out &
 nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $KEGG --annotation-file $GTF_KEGG --simulations 100000 --min-significance 1 --gene-definition gene --threads 2 --output-file `basename $SIGN .cmhout`"_gene_KEGG.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_gene_KEGG.gowinda".out &
 nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $REACTOME --annotation-file $GTF_REACTOME --simulations 100000 --min-significance 1 --gene-definition gene --threads 2 --output-file `basename $SIGN .cmhout`"_gene_REACTOME.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_gene_REACTOME.gowinda".out &
 nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $RAY_ASSOX --annotation-file $GTF_CESAR --simulations 100000 --min-significance 1 --gene-definition gene --threads 2 --output-file `basename $SIGN .cmhout`"_gene_RAY_ASSOX.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_gene_RAY_ASSOX.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $GO --annotation-file $GTF_GO --simulations 100000 --min-significance 1 --gene-definition exon --threads 2 --output-file `basename $SIGN .cmhout`"_exon_GO.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_exon_GO.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $KEGG --annotation-file $GTF_KEGG --simulations 100000 --min-significance 1 --gene-definition exon --threads 2 --output-file `basename $SIGN .cmhout`"_exon_KEGG.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_exon_KEGG.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $REACTOME --annotation-file $GTF_REACTOME --simulations 100000 --min-significance 1 --gene-definition exon --threads 2 --output-file `basename $SIGN .cmhout`"_exon_REACTOME.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_exon_REACTOME.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $RAY_ASSOX --annotation-file $GTF_CESAR --simulations 100000 --min-significance 1 --gene-definition exon --threads 2 --output-file `basename $SIGN .cmhout`"_exon_RAY_ASSOX.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_exon_RAY_ASSOX.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $GO --annotation-file $GTF_GO --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_GO.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_GO.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $KEGG --annotation-file $GTF_KEGG --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_KEGG.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_KEGG.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $REACTOME --annotation-file $GTF_REACTOME --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_REACTOME.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_REACTOME.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $RAY_ASSOX --annotation-file $GTF_CESAR --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_RAY_ASSOX.gowinda" --mode gene --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_RAY_ASSOX.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $GO --annotation-file $GTF_GO --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_snp_GO.gowinda" --mode snp --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_snp_GO.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $KEGG --annotation-file $GTF_KEGG --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_snp_KEGG.gowinda" --mode snp --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_snp_KEGG.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $REACTOME --annotation-file $GTF_REACTOME --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_snp_REACTOME.gowinda" --mode snp --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_snp_REACTOME.gowinda".out &
nohup java -Xmx4g -jar $GOWINDA --snp-file $TOTAL --candidate-snp-file $SIGN --gene-set-file $RAY_ASSOX --annotation-file $GTF_CESAR --simulations 100000 --min-significance 1 --gene-definition upstream1000 --threads 2 --output-file `basename $SIGN .cmhout`"_upstream1000_snp_RAY_ASSOX.gowinda" --mode snp --min-genes 1 2>&1 >>  `basename $SIGN .cmhout`"_upstream1000_snp_RAY_ASSOX.gowinda".out &
